slides/C-concurrency-parallelism.md

```diff
@@ -0,0 +1,606 @@
+---
+theme: default
+class: text-center
+highlighter: shiki
+lineNumbers: true
+info: "Rust - X: Y"
+drawings:
+  persist: false
+fonts:
+  mono: Fira Mono
+layout: cover
+title: 'Rust - X: Y'
+---
+# Rust programming
+Module C: concurrency and parallelism 
+<!-- Start with welcome, students entering -->
+<!-- TODO add subject code -->
+
+---
+layout: three-slots
+---
+## Who am i?
+::left::
+- Ferris
+- I Love Rust
+
+::right::
+<img src="https://rustacean.net/assets/rustacean-orig-noshadow.png" alt="Photo Ferris" width="300" />
+<!-- Optionally quickly introduce yourself, add photo -->
+
+---
+layout: default
+---
+# Last time...
+- Ownership model
+- Move semantics
+<!-- Recap on content from last time that current subject builds on -->
+
+
+---
+layout: default
+---
+# Closures 
+
+- Closures are anonymous (unnamed) functions 
+- they can capture ("close over") values in their scope
+- they are first-class values
+
+```rust
+fn foo() -> impl Fn(i64, i64) -> i64 {
+    z = 42;
+    |x, y| x + y + z
+}
+
+fn bar() -> i64 {
+    // construct the closure
+    let f = foo();
+
+    // evaluate the closure
+    f(1, 2)
+}
+```
+
+- very useful when working with iterators, `Option` and `Result`.
+
+```rust
+let evens: Vec<_> = some_iterator.filter(|x| x % 2 == 0).collect();
+```
```

Comment:
Maybe do this after the learning objectives

Reply:
yeah well this does not fit in anywhere... I'll play around with it